### PR TITLE
Add a way to extend the UI with an Java API

### PR DESCRIPTION
**Pull request by edewit**
Initial version of the API that enables java developers to extend the admin UI

### CODE REVIEW

**Comment by hypery2k** [comment]
would love to see this feature merged :D

File: common/src/main/java/org/keycloak/common/Profile.java [hunk 1]
```diff
@@ -104,6 +104,8 @@ public enum Feature {
         MULTI_SITE("Multi-site support", Type.PREVIEW),
 
         OFFLINE_SESSION_PRELOADING("Offline session preloading", Type.DEPRECATED),
+
+        UI_CUSTOMIZATION("UI customization spi", Type.EXPERIMENTAL),
```
Comment (ssilvert) [inline common/src/main/java/org/keycloak/common/Profile.java:108]: Didn't we have another name for this?  Maybe it was something like, "Declarative UI?"

UI_CUSTOMIZATION is too broad of a term to describe what this does as there are other ways to customize the UI.

File: js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties [hunk 2]
```diff
@@ -2985,4 +2985,16 @@ readBlog=Read blog
...
+createItem=Create item
+itemDelete=Delete item
+itemDeleteConfirm=Are you sure you want to permanently delete the item
+itemDeleteConfirmTitle=Delete item?
+itemDeletedSuccess=The item has been deleted
+itemDeleteError=Could not delete item: {{error}}
+noItems=There are no items
+noItemsInstructions=You haven't created any items in this realm. Create a item to get started.
+itemSaveError=Error could not save item\! {{error}}
+itemSaveSuccessful=Sucessful saved
```
Comment (ssilvert) [inline js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:3000]: I wouldn't change this for this first version, but in the future a developer would want to use the actual name of the item.

So for instance, `searchItem=Search item` would become `searchItem=Search {{itemName}}`

**Review by ssilvert — CHANGES_REQUESTED** [review]

**Review by ssilvert — APPROVED** [review]

**Comment by smandlavdiya** [comment]
> Initial version of the API that enables java developers to extend the admin UI

I was trying to add a new tab on client but unable to figure out how would i get client scope when i save and add values with oncreate()
any suggestions would be good to start please

**Comment by carlosrodovalho** [comment]
I am experiencing an issue with my implementation of the ThemeUiTab class. The implementation works correctly when updating the realm attribute using the administrative interface. However, when I use the REST API to update the realm attribute, the new attribute is saved (as confirmed by checking Keycloak's database), but the input field in the administrative interface still displays the old value.

I have tried reloading Keycloak, but the interface does not load the new value from the database.

For reference, I implemented the example from the following repository: 
https://github.com/keycloak/keycloak-quickstarts/blob/latest/extension/extend-admin-console-spi/src/main/java/org/keycloak/admin/ui/ThemeUiTab.java

**Pull request merged**